sdk/azidentity/bearer_token_policy_test.go

```diff
@@ -174,15 +176,22 @@ func TestBearerPolicy_GetTokenFailsNoDeadlock(t *testing.T) {
 		HTTPClient:    srv,
 		AuthorityHost: srv.URL(),
 		Retry: azcore.RetryOptions{
-			// leaving TryTimeout at zero will trigger a deadline exceeded error causing GetToken() to fail
-			RetryDelay:    50 * time.Millisecond,
-			MaxRetryDelay: 500 * time.Millisecond,
-			MaxRetries:    3,
+			// leaving TryTimeout at as a very small value will trigger a deadline exceeded error causing GetToken() to fail
+			TryTimeout: 1 * time.Nanosecond,
+			MaxRetries: -1,
 		}})
 	if err != nil {
 		t.Fatalf("Unable to create credential. Received: %v", err)
 	}
-	pipeline := defaultTestPipeline(srv, cred, scope)
+	pipeline := azcore.NewPipeline(
+		srv,
+		azcore.NewRetryPolicy(&azcore.RetryOptions{
+			MaxRetryDelay: 500 * time.Millisecond,
+			RetryDelay:    50 * time.Millisecond,
+			MaxRetries:    -1,
```

Comment:
This also disables retries.  Can it be switched back to `defaultTestPipeline()`?

Reply:
This is the same test as above. I'm looking into why this test is broken.

Reply:
Oh, I also agree it was wrong to put MaxRetries at -1. I'll make sure the fix doesnt have that.

Reply:
Left this using the defaultTestPipeline(). Done.